src/neo/Persistence/StoreView.cs

```diff
@@ -16,13 +16,16 @@ public abstract class StoreView
         public abstract DataCache<UInt160, ContractState> Contracts { get; }
         public abstract DataCache<StorageKey, StorageItem> Storages { get; }
         public abstract DataCache<SerializableWrapper<uint>, HeaderHashList> HeaderHashList { get; }
+        public abstract DataCache<SerializableWrapper<uint>, HashIndexState> LocalStateRoot { get; }
         public abstract MetaDataCache<HashIndexState> BlockHashIndex { get; }
         public abstract MetaDataCache<HashIndexState> HeaderHashIndex { get; }
+        public abstract MetaDataCache<StateRoot> ValidatorsStateRoot { get; }
```

Comment:
What's the difference between `LocalStateRoot` and `ValidatorsStateRoot`?

Reply:
`LocalStateRoot` is calculated by local node.
`ValidatorsStateRoot` is the StateRoot with signatures of validators.

Reply:
So we don't need `LocalStateRoot`.

Reply:
We need last state root to get storage and calculate current state root when persist block.
Maybe use `CurrentLocalStateRoot`?
If we use `ValidatorsStateRoot` directly, node can't sync block when state root is different from `ValidatorsStateRoot` or not complete sync. Is this acceptable?

Reply:
We need `LocalStateRoot` to decouple block sync and state root sync.